File: socialuni-center-web/src/main/java/com/socialuni/center/web/feignAPI/SocialuniLocationAPI.java
package com.socialuni.center.web.feignAPI;

import com.socialuni.center.web.model.QO.location.LocationQueryQO;
import com.socialuni.center.web.model.QO.location.LocationQueryRO;
import com.socialuni.social.api.model.ResultRO;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;

import javax.validation.Valid;

@RequestMapping("location")
@FeignClient(name = "location", url = "${socialuni.server-url:https://api.socialuni.cn}")
public interface SocialuniLocationAPI {
    @PostMapping("queryLocation")
    ResultRO<LocationQueryRO> queryLocation(@RequestBody @Valid LocationQueryQO queryQO);
}

File: socialuni-js/packages/socialuni-app/socialuni-app-api/src/api/SocialuniLocationAPI.ts
import LocationQueryQO from "socialuni-api-base/src/model/location/LocationQueryQO";
import {QQMapResult} from "socialuni-api-base/src/model/location/QQMapResult";
import ResultRO from "socialuni-api-base/src/model/social/ResultRO";
import socialuniAppRequest from "../socialuniAppRequest";

export default class SocialuniLocationAPI {
    static queryLocation(location: LocationQueryQO = new LocationQueryQO()): Promise<ResultRO<QQMapResult>> {
        //小程序，且注册了qq的地图key
        return socialuniAppRequest.post<QQMapResult>('socialuni/location/queryLocation', location) as any
    }
}
